src/core/parser.zsh

```diff
@@ -111,24 +115,52 @@ _fzf_complete_parse_argument() {
 }
 
 _fzf_complete_parse_option() {
-    local result=()
     local shorts=(${(z)1})
     local longs=(${(z)2})
     local options_argument_required=(${(z)3})
     shift 3
 
-    local cmd=("${(Q)${(z)@}[@]}")
-    local cmd_shorts=(${(M)cmd:#-[^-]*})
-
-    local option_argument_required
-    for option_argument_required in $options_argument_required; do
-        cmd_shorts=(${cmd_shorts%%${option_argument_required#-}*})
-    done
+    local i j parsing_argument
+    local result=()
+    local start_index=2
+    local arguments=("$@")
 
-    cmd_shorts=(-${^${(ps::)cmd_shorts}})
+    for i in {$start_index..${#arguments}}; do
+        if [[ -n $parsing_argument ]]; then
+            parsing_argument=
+            continue
+        fi
 
-    result+=(${cmd_shorts:*shorts})
-    result+=(${cmd:*longs})
+        if [[ ${arguments[$i]} = -- ]]; then
+            break
+        elif [[ ${arguments[$i]} = -[A-Za-z0-9]* ]]; then
+            for j in {$start_index..${#arguments[$i]}}; do
+                if [[ ${options_argument_required[(r)-${arguments[$i][$j]}]} = -${arguments[$i][$j]} ]]; then
+                    if [[ $j = ${#arguments[$i]} ]]; then
+                        parsing_argument=1
+                    else
+                        parsing_argument=
+                        break
+                    fi
+                elif [[ ${shorts[(r)-${arguments[$i][$j]}]} = -${arguments[$i][$j]} ]]; then
+                    if [[ $j = ${#arguments[$i]} ]]; then
+                        if (( ${#arguments} = $i )); then
+                            break
+                        fi
+                        parsing_argument=
+                    fi
+                    result+=(-${arguments[$i][$j]})
+                fi
+            done
+        elif [[ ${arguments[$i]} = --[A-Za-z0-9]* ]]; then
```

Comment:
This is the same above.
```suggestion
        elif [[ ${arguments[$i]} = --[^-]* ]]; then
```
Alternatively, it may be the following:
```suggestion
        elif [[ ${arguments[$i]} = --* ]]; then
```

Reply:
Cool ✨